src/common/transformations/src/transformations/resolve_names_collisions.cpp

```diff
@@ -43,19 +43,38 @@ bool ov::pass::ResolveNameCollisions::run_on_model(const std::shared_ptr<ov::Mod
             continue;
         }
 
-        int64_t cnt = 2;
-        for (const auto& op : same_named_ops) {
-            // check if op has OV autogenerated friendly name. unique and friendly names have to be equal.
-            bool is_autogenerated = op->m_friendly_name.empty();
-            if (!is_autogenerated) {
-                continue;
-            }
+        auto update_name = [&](ov::Node* op, int64_t& cnt) {
             // add a prefix "_counter" to the autogenerated name to make it unique.
-            auto new_name = op->get_friendly_name() + "_" + std::to_string(cnt++);
+            const auto& old_name = op->get_friendly_name();
+            auto new_name = old_name + "_" + std::to_string(cnt++);
             while (visited_nodes.find(new_name) != visited_nodes.end()) {
```

Comment:
should not we update a map of `visited_nodes` with a new name?

Reply:
Good point! I believe we don't need and that's why:
Let's say  "visited_nodes" map initially contains
[prefix1] = node1, node2, node3
[prefix2] = node4, node5
, where prefix1 != prefix2

After applying the transformation for "prefix1" and adding the new elements to "visited_nodes" map
[prefix1] = node1
[prefix2] = node4, node5
[prefix3 = prefix1 + id] = node2
[prefix4 = prefix1 + id] = node3
where prefix3 != prefix4 and prefix3 != prefix2, prefix4 != prefix2

The question is: is it possible to get prefix1 + id from prefix2 by adding id?
I don't think so, because prefix1 != prefix2